core/src/main/scala/kafka/admin/PreferredReplicaLeaderElectionCommand.scala

```diff
@@ -38,41 +48,50 @@ object PreferredReplicaLeaderElectionCommand extends Logging {
+    val zkConnectOpt = parser.accepts("zookeeper", "DEPRECATED. The connection string for the zookeeper connection in the " +
+      "form host:port. Multiple URLS can be given to allow fail-over. REQUIRED unless --bootstrap-server is given.")
       .withRequiredArg
       .describedAs("urls")
       .ofType(classOf[String])
-      
+    val bootstrapServerOpt = parser.accepts("bootstrap-server", "A hostname and port for the broker to connect to, " +
+      "in the form host:port. Multiple comma-separated URLs can be given. REQUIRED unless --zookeeper is given.")
+      .withRequiredArg
+      .describedAs("host:port")
+      .ofType(classOf[String])
+
+
     if(args.length == 0)
-      CommandLineUtils.printUsageAndDie(parser, "This tool causes leadership for each partition to be transferred back to the 'preferred replica'," + 
-                                                " it can be used to balance leadership among the servers.")
+      CommandLineUtils.printUsageAndDie(parser, "This tool causes leadership for each partition to be transferred back to the 'preferred replica'," +
+                                                " it can be used to balance leadership among the servers." +
+                                                " The preferred replica is the first one in the replica assignment (see kafka-reassign-partitions.sh)." +
+                                                " Using this command is not necessary when the broker is configured with \"auto.leader.rebalance.enable=true\".")
 
     val options = parser.parse(args : _*)
 
-    CommandLineUtils.checkRequiredArgs(parser, options, zkConnectOpt)
+    if (options.has(bootstrapServerOpt) == options.has(zkConnectOpt)) {
+      CommandLineUtils.printUsageAndDie(parser, s"Exactly one of '${bootstrapServerOpt}' or '${zkConnectOpt}' must be provided")
+    }
 
-    val zkConnect = options.valueOf(zkConnectOpt)
-    var zkClient: KafkaZkClient = null
-    try {
-      val time = Time.SYSTEM
-      zkClient = KafkaZkClient(zkConnect, JaasUtils.isZkSecurityEnabled, 30000, 30000, Int.MaxValue, time)
+    val partitionsForPreferredReplicaElection =
+    if (options.has(jsonFileOpt))
+      Some(parsePreferredReplicaElectionData(Utils.readFileAsString(options.valueOf(jsonFileOpt))))
+    else
+      None
 
-      val partitionsForPreferredReplicaElection =
-        if (!options.has(jsonFileOpt))
-          zkClient.getAllPartitions()
-        else
-          parsePreferredReplicaElectionData(Utils.readFileAsString(options.valueOf(jsonFileOpt)))
-      val preferredReplicaElectionCommand = new PreferredReplicaLeaderElectionCommand(zkClient, partitionsForPreferredReplicaElection)
+    val preferredReplicaElectionCommand = if (options.has(zkConnectOpt)) {
+      Console.err.println(s"$zkConnectOpt is deprecated and will be removed in a future version of Kafka.")
+      Console.err.println(s"Use $bootstrapServerOpt instead to specify a broker to connect to.")
```

Comment:
Could we print this as a warning? Also, instead of printing to Console.err, perhaps just print to stdout?